SCons/Environment.py

```diff
@@ -2000,6 +2001,19 @@ def CacheDir(self, path):
 
     def Clean(self, targets, files):
         global CleanTargets
+
+        # Check for anything which evaluates to empty string, which would yield cleaning '.'
+        targets_strings = [(t, self.subst(t)) for t in flatten(targets) if is_String(t) and ('$' in t or t == '')]
+        files_strings = [(t, self.subst(t)) for t in flatten(files) if is_String(t) and ('$' in t or t == '')]
+        if any([s == '' for t, s in targets_strings]):
+            raise UserError(
+                "Targets specified to Clean() include one which evaluates to an empty string: [%s]" % ",".join(
+                    ["%s='%s'" % (str(t), s) for (t, s) in targets_strings]))
+        if any([s == '' for t, s in files_strings]):
```

Comment:
I see that one stanza checks targets_strings and the other files_strings, but this still reads quite duplicative... any way to combine?